### ISSUE TITLE
Exception with Python 3.9 - unsupported operand type

### ISSUE THREAD

**Issue by ReneNulschDE** [mampfes/hacs_waste_collection_schedule#334]
Hi,

Thanks for this great custom component. 

I'm running latest HA core version in an virtual environment with Python 3.9. (Still supported by HA)

I get the following exception on startup:
```
File "/home/rene/.homeassistant/custom_components/waste_collection_schedule/calendar.py", line 118, in <module>
    def calc_unique_calendar_id(scraper: Scraper, type: str | None = None):
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
2022-08-17 19:16:20.603 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform waste_collection_schedule.calendar: Platform not found (Exception importing custom_components.waste_collection_schedule.calendar).
```

Looks like the operand "|" is not supported in Python3.9.
Pylance message: "Alternative syntax for unions requires Python 3.10"

**Comment by dt215git** [comment]
I think this was discussed in #316

**Comment by mampfes** [comment]
fixed with #346
